wandb/sdk/service/streams.py

```diff
@@ -245,9 +248,21 @@ def _process_action(self, action: StreamAction) -> None:
             return
         raise AssertionError(f"Unsupported action: {action._action}")
 
+    def _check_orphaned(self) -> bool:
+        if self._pid is None:
+            return False
+        time_now = time.time()
+        # if we have checked already and it was less than 2 seconds ago
+        if self._pid_checked_ts and time_now < self._pid_checked_ts + 2:
+            return False
+        self._pid_checked_ts = time_now
+        return not psutil.pid_exists(self._pid)
+
     def _loop(self) -> None:
         while not self._stopped.is_set():
-            # TODO: check for parent process going away
+            if self._check_orphaned():
```

Comment:
is it always the case if the parent process goes away we have to shut it down?

Reply:
I think so...
There is a question if we should try to do some cleanup though, like finish syncing data.
For this release I thought this was a good tradeoff between just calling `os._exit()` and trying to let things shut down a bit more gracefully.

If the parent process went away, then control was returned to the caller process so it is expected that everything is committed that should be saved.  If we continued to do work we end up creating a situation where the data might be different if they queried immediately vs 5 seconds later.   So ideally we find ways to be even more precise in shutting down the wandb service process.   There are some tricks that we can do on linux which will do this at the OS layer.